### ISSUE TITLE
LOADBG with AUTO immediately fires in progressive video-mode (2.1.0)

### ISSUE THREAD

**Issue by pkeuter** [CasparCG/Server#467]
When a channel is configured in a progressive video-mode, the LOADBG command immediately fires and does not wait until the end of the currently playing video. 

I have tested this with PAL, 720p5000, 1080i5000 and 1080p5000. The commands I have used are:

```
PLAY 1-1 AMB
LOADBG 1-1 EMPTY AUTO 
```

In PAL and 1080i5000 the video ends and the ffmpeg producer is destroyed.

```
[2016-08-21 12:11:40.047] [976]   [info]    color[empty] Initialized
[2016-08-21 12:11:40.050] [976]   [info]    transition[empty=>color[empty]] Initialized
[2016-08-21 12:11:44.842] [7528]  [info]    ffmpeg[AMB.mp4|1920x1080p25.00|267/268] Destroyed.
```

In 720p5000 and 1080p5000 the ffmpeg producer is immediately destroyed:

```
[2016-08-21 12:09:54.123] [2152]  [info]    color[empty] Initialized
[2016-08-21 12:09:54.124] [2152]  [info]    transition[empty=>color[empty]] Initialized
[2016-08-21 12:09:54.143] [15136] [info]    ffmpeg[AMB.mp4|1920x1080p25.00|104/268] Destroyed.
```

You can check the timestamps to understand what I mean.

**Comment by pkeuter** [comment]
Is there anyone here who can give me a lead so I know where to start searching?

**Comment by pkeuter** [comment]
This issue must have something to do with the framerate producer and output_repeat.

I've been able to fix this issue by changing the following lines:
```

	uint32_t nb_frames() const override
	{
		return static_cast<uint32_t>((source_->nb_frames() * boost::rational_cast<double>(1 / get_speed()) * (output_repeat_ != 0 ? 2 : 1)));
	}

	uint32_t frame_number() const override
	{
		return static_cast<uint32_t>((source_->frame_number() * boost::rational_cast<double>(1 / get_speed()) * (output_repeat_ != 0 ? 2 : 1)));
	}
```

But I am not sure if I'm breaking stuff here. It would seem logical to me that when frames are duplicated (interlaced input, progressive output), the calls to nb_frames and frame_number should be doubled instead of halved, right?

I'm not sure if this has implications when altering the playback speed since this doesn't seem to work in the latest build, but maybe I'm doing something wrong here.

So if this doesn't break anything, @HellGore, @ronag, I will file a PR.

**Comment by HellGore** [comment]
I am already working on this, there are a number of related bugs that I need to fix at the same time to make it work.

**Comment by ronag** [comment]
should be fixed in 2.2.0
